### ISSUE TITLE
How to Implement This in New Scene

### ISSUE THREAD

**Issue by itsNaz** [SebLague/Clouds#5]
I have been trying to make a new scene using these clouds (so I could tailor them to my scenes) and I found out that the scene stays blank. This persists even if I copy and paste the test scene. I am not quite sure if this is a bug or not. #

**Comment by theishiopian** [comment]
Necro here, but on the off chance it helps, be sure to copy the cloud master script from the main camera on to your camera and set the container transform value in the inspector

**Comment by ZeusMoo** [comment]
I was wondering if anyone else had luck getting this to work as I had the same issues.  I could temporarily get the clouds to show up in one of my scenes if I imported and ran one of the included test scenes, but the clouds eventually disappear and they don't show up at all on export.

**Comment by theishiopian** [comment]
Yeah I have similar issues. I just bought a cloud asset after a while. After all, this is an experiment, not a commercial asset.

**Comment by itsNaz** [comment]
That's annoying, but to be expected I guess. I moved on from this little project (it was for school), but thanks for the tip theishiopian.

**Comment by Cytochrome8942** [comment]
Same here. After 3 hours of experiment, I duplicated Airplane test scene and added all of my original scene's objects. It worked like 5 minutes, but after that they disappeared. I couldn't find the problem, so I quit.

**Comment by Cytochrome8942** [comment]
https://github.com/yangrc1234/VolumeCloud
I used this package.

**Comment by theishiopian** [comment]
That package is a good one, but it tends to interfere with transparent materials.

**Comment by AB-DHBW-LOE** [comment]
if sb still is having issues:
managed to create new scenes, by copying all objects to my scene and in the NoiseGenerator.cs file changing 
`Load` and `Save` to load a static 3d-texture file instead of one which depends on the scene-name.

Setting *all* parameters exactly the same is quite important (atleast for most parameters) otherwise the clouds tend to not even show up at all. (don't forget camera script & container!)

**Comment by elliot-winch** [comment]
> if sb still is having issues: managed to create new scenes, by copying all objects to my scene and in the NoiseGenerator.cs file changing `Load` and `Save` to *load a static 3d-texture file instead of one which depends on the scene-name.*

I was searching for ages trying to make this work in a new scene for a side project. Thank you for posting this, it saved so much time!

Specifically, the dependency on scene name is at NoiseGenerator.cs Line 108:
        string sceneName = UnityEngine.SceneManagement.SceneManager.GetActiveScene ().name;

You need to create Assets > Resources > {Scene Name}_DetailsNoise and  {Scene Name}_DetailsNoise, most easily by duplicating and renaming the existing textures.

**Comment by PakingPie** [comment]
I explore the compute shader a bit and find that the Result[i] always returns 0. However, if you can pre-fill the input texture with a random number, the final Texture3D will be properly generated.
Here is the code: 
  ```
//
  //
  // Pre-fill the shapeTexture with random values to initialize
  if (ActiveTexture != null)
  {
      Texture3D tempTex = new Texture3D(ActiveTexture.width, ActiveTexture.height, ActiveTexture.volumeDepth, TextureFormat.RGBAFloat, false);
      Color[] colors = new Color[ActiveTexture.width * ActiveTexture.height * ActiveTexture.volumeDepth];
      for (int x = 0; x < ActiveTexture.width; x++)
      {
          for (int y = 0; y < ActiveTexture.height; y++)
          {
              for (int z = 0; z < ActiveTexture.volumeDepth; z++)
              {
                  int idx = x + ActiveTexture.width * (y + z * ActiveTexture.height);
                  colors[idx] = Random.ColorHSV().linear;
              }
          }
      }
      tempTex.SetPixels(colors);
      tempTex.Apply();
  
      // Copy Texture3D to RenderTexture using compute shader if needed, or skip this step if not required
      // For now, just clean up tempTex
      // convert the tempTex to the RenderTexture
      copy.SetTexture(0, "tex", tempTex);
      copy.SetTexture(0, "renderTex", ActiveTexture);
      copy.Dispatch(0, Mathf.CeilToInt(ActiveTexture.width / 8f), Mathf.CeilToInt(ActiveTexture.height / 8f), Mathf.CeilToInt(ActiveTexture.volumeDepth / 8f));
      Object.DestroyImmediate(tempTex);
  }
```
